sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableWriteIT.java

```diff
@@ -85,17 +88,18 @@ public void setup() throws Exception {
     retryOptionsBuilder.setStreamingBatchSize(
         retryOptionsBuilder.build().getStreamingBufferSize() / 2);
 
-    BigtableOptions.Builder bigtableOptionsBuilder =
-        new BigtableOptions.Builder()
-            .setProjectId(options.getProjectId())
-            .setInstanceId(options.getInstanceId())
-            .setUserAgent("apache-beam-test")
-            .setRetryOptions(retryOptionsBuilder.build())
-            .setCredentialOptions(
-                CredentialOptions.credential(options.as(GcpOptions.class).getGcpCredential()));
-    bigtableOptions = bigtableOptionsBuilder.build();
+    Credentials credentials = new GcpUserCredentialsFactory().create(options);
```

Comment:
It still makes sense to use the credentials from pipeline options instead of using the factory since pipeline options might be configured with something like a service account location or something else.